nixos/modules/services/hardware/intel-lpmd.nix

```
@@ -0,0 +1,527 @@
+          Set both this and `util_entry_threshold` to '0' to disable this feature.
+        '';
+
+        type = with lib.types; numbers.between 0 100;
+      };
+
+      EntryDelayMS = lib.mkOption {
+        default = 0;
+        description = ''
+          Specifies the sample interval (in ms) used by the utilization monitor
+          when system wants to enter LPM.
+
+          Setting this to '0' uses the default interval of 1000ms.
+        '';
+
+        type = with lib.types; ints.unsigned;
+      };
+
+      ExitDelayMS = lib.mkOption {
+        default = 0;
+        description = ''
+          Specifies the sample interval (in ms) used by the utilization monitor
+          when system wants to exit LPM.
+
+          Setting this to '0' uses an adaptive value which is calculated based
+          on CPU utilization.
+        '';
+
+        type = with lib.types; ints.unsigned;
+      };
+
+      # TODO: it is unclear if disabling requires both to be set to '0' or not.
+      EntryHystMS = lib.mkOption {
+        default = "0";
+        description = ''
+          Specifies a hysteresis threshold (in ms) when the system is in LPM.
+          When the previous average time **stayed in** LPM is lower than this
+          value, the **current enter** LPM request will be ignored, due to the
+          expectation of the system exiting LPM soon.
+
+          Setting this to '0' disables the hysteresis algorithm.
+        '';
+
+        type = with lib.types; ints.unsigned;
+      };
+
+      ExitHystMS = lib.mkOption {
+        default = "0";
+        description = ''
+          Specifies a hysteresis threshold (in ms) when the system is not in LPM.
+          When the previous average time **stayed out of** LPM is lower than this
+          value, the **current exit** LPM request will be ignored, due to the
+          expectation of the system entering LPM soon.
+
+          Setting this to '0' disables the hysteresis algorithm.
+        '';
+
+        type = with lib.types; ints.unsigned;
+      };
+
+      lp_mode_epp = lib.mkOption {
+        default = null;
+        description = ''
+          EPP (energy performance preferences) to use in LPM.
+          Leave as `null` to ignore this setting.
+
+          Accepts a value from **0** (favor performance)
+          to **255** (favor power).
+        '';
+
+        type = with lib.types; nullOr numbers.between 0 255;
+      };
+
+      IgnoreITMT = lib.mkOption {
+        default = false;
+        description = ''
+          Specifies whether to avoid changing the scheduler ITMT flag.
+
+          If enabled, ITMT is disabled upon entering LPM,
+          and re-enabled upon exiting LPM. Otherwise, the
+          ITMT setting is ignored entirely.
+        '';
+      };
+
+      States = lib.mkOption {
+        default = [ ];
+        description = ''
+          List of per-platform low power states.
+        '';
+
+        type = with lib.types; listOf (submodule statesSubmodule);
```

```suggestion
      States = lib.mkOption {
        type = with lib.types; listOf (submodule statesSubmodule);
        default = [ ];
        description = "List of per-platform low power states.";
```
Can you put all the type arguments in the beginning of mkOption and not with an extra new line in the end? It could be easily missed there and is generally not how we write things.